homeassistant/components/zimi/light.py

```diff
@@ -0,0 +1,116 @@
+"""Platform for light integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from zcc import ControlPoint
+from zcc.device import ControlPointDevice
+
+from homeassistant.components.light import ATTR_BRIGHTNESS, ColorMode, LightEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+
+# Import the device class from the component that you want to support
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Zimi Light platform."""
+
+    api: ControlPoint = config_entry.runtime_data
+
+    async_add_entities(ZimiLight(device, api) for device in api.lights)
+
+
+class ZimiLight(LightEntity):
+    """Representation of a Zimi Light."""
+
+    def __init__(self, light: ControlPointDevice, api: ControlPoint) -> None:
+        """Initialize an ZimiLight."""
+
+        self._attr_unique_id = light.identifier
+        self._attr_name = light.name.strip()
+        self._attr_should_poll = False
+        self._light = light
+        self._light.subscribe(self)
+        if self._light.type == "dimmer":
+            self._attr_color_mode = ColorMode.BRIGHTNESS
+            self._attr_supported_color_modes = {ColorMode.BRIGHTNESS}
+        else:
+            self._attr_color_mode = ColorMode.ONOFF
+            self._attr_supported_color_modes = {ColorMode.ONOFF}
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, light.identifier)},
+            name=self._attr_name,
+            suggested_area=self._light.room,
+            via_device=(DOMAIN, api.mac),
+        )
+        _LOGGER.debug("Initialising %s in %s", self.name, self._light.room)
+
+    @property
+    def available(self) -> bool:
+        """Return True if Home Assistant is able to read the state and control the underlying device."""
+        return self._light.is_connected
+
+    @property
+    def brightness(self) -> int | None:
```

Comment:
I'm thinking if it makes sense to define a `ZimiLight` and a `ZimiDimmer(ZimiLight)` so we can move the brightness stuff to the dimmer  only

Reply:
Done https://github.com/home-assistant/core/pull/129876/commits/f8d4ba0a116887413d4b503ab4abde43564b512a